File: backend/app/database/models/document.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import Column, String, Integer, BigInteger, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.database.base import Base

class Document(Base):
    __tablename__ = "documents"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    contract_id = Column(String(36), ForeignKey("contracts.id", ondelete="CASCADE"), nullable=False, index=True)
    
    document_type = Column(String(50), nullable=False)  # BASELINE, AMENDMENT, INVOICE, PROGRESS_REPORT
    version_number = Column(Integer, nullable=True)
    
    filename = Column(String(255), nullable=False)
    storage_path = Column(String(500), nullable=False)
    mime_type = Column(String(100), nullable=True)
    file_size = Column(BigInteger, nullable=True)
    
    processing_status = Column(String(50), nullable=False, default="UPLOADED")  # UPLOADED, PROCESSING, COMPLETED, FAILED
    uploaded_by = Column(String(36), nullable=True)
    
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))
    updated_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), onupdate=lambda: datetime.now(timezone.utc))

    # Relationships
    contract = relationship("Contract", back_populates="documents")
    version = relationship("ContractVersion", back_populates="document", uselist=False)

File: frontend/src/types/document.ts
export type DocumentType = 'BASELINE' | 'AMENDMENT' | 'INVOICE' | 'PROGRESS_REPORT';

export type ProcessingStatus = 'UPLOADED' | 'PROCESSING' | 'COMPLETED' | 'FAILED';

export interface Document {
  id: string;
  contract_id: string;
  document_type: DocumentType;
  version_number: number | null;
  filename: string;
  storage_path: string;
  mime_type: string | null;
  file_size: number | null;
  processing_status: ProcessingStatus;
  uploaded_by: string | null;
  created_at: string;
  updated_at: string;
}

export interface DocumentDownloadUrl {
  document_id: string;
  download_url: string;
  filename: string;
  expires_in_seconds: number;
}
